app/views/curriculum/units/_lessons.html.erb

```diff
@@ -3,7 +3,19 @@
 		<h3 class="govuk-heading-m">Lessons</h3>
 			<ul class="govuk-list govuk-body govuk-!-padding-bottom-7">
         <% lessons.each do |lesson| %>
-            <li class="curriculum__list--item <%= key_stage_list_color(key_stage.level) %>"><%= link_to lesson_title_wording(lesson), curriculum_key_stage_unit_lesson_path(key_stage_slug: key_stage.slug, unit_slug: unit.slug, lesson_slug: lesson.slug), class: 'govuk-link ncce-link' %></li>
+					<li class="curriculum__list--item curriculum__list--item-with-download <%= key_stage_list_color(key_stage.level) %>">
+						<%= link_to lesson_title_wording(lesson), curriculum_key_stage_unit_lesson_path(key_stage_slug: key_stage.slug, unit_slug: unit.slug, lesson_slug: lesson.slug), class: 'govuk-link ncce-link' %>
+
+						<% if current_user %>
+							<% if lesson.zipped_contents&.file.present? %>
```

Comment:
Would we want to rearrange these conditionals slightly? This way round we could end up telling someone to log in and then once they log in there's nothing for them to download. Perhaps the `file.present?` check is around the whole block and then the logged in yes/no branches inside that?

Reply:
Makes a lot of sense, will make those changes, thanks!